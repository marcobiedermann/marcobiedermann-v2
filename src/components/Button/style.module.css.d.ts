/* eslint-disable import/prefer-default-export */

export const button: string;
