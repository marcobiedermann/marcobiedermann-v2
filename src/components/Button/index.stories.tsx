/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Button from '.';

storiesOf('Button', module)
  .add('default', () => <Button>Content</Button>)
  .add('link', () => <Button href="#">Content</Button>);
