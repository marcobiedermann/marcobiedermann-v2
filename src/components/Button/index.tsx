import classNames from 'classnames';
import { Link } from 'gatsby';
import React from 'react';
import styles from './style.module.css';

export interface ButtonProps {
  className?: string;
  href?: string;
  to?: string;
}

const Button: React.FC<ButtonProps> = (prop) => {
  const { children, className, href, to } = prop;
  const classes = classNames(className, styles.button);

  if (href) {
    return (
      <a href={href} className={classes} target="_blank" rel="noopener noreferrer">
        {children}
      </a>
    );
  }

  if (to) {
    return (
      <Link className={classes} to={to}>
        {children}
      </Link>
    );
  }

  return <span className={classes}>{children}</span>;
};

export default Button;
