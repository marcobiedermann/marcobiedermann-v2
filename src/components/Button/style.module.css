.button {
  border: 1px solid #0a0a0a;
  color: #0a0a0a;
  display: inline-block;
  overflow: hidden;
  padding: 0.5em 1em;
  position: relative;
  text-decoration: none;
}

.button.focus,
.button:hover {
  color: #f8f9fb;
}

.button::after {
  background-color: #0a0a0a;
  content: '';
  height: 375%;
  left: 50%;
  opacity: 0;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%) rotate(45deg) scaleY(0);
  transition: transform 0.3s, opacity 0.3s;
  width: 100%;
  z-index: -1;
}

.button.focus::after,
.button:hover::after {
  opacity: 1;
  transform: translate(-50%, -50%) rotate(45deg) scaleY(1);
}
