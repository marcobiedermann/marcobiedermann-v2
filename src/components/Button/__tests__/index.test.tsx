/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Button from '..';

describe('Button component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render link using `href`', () => {
    const { asFragment } = render(<Button href="#" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render link using `to`', () => {
    const { asFragment } = render(<Button to="#" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
