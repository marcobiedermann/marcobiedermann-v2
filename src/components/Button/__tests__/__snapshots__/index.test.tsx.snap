// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<DocumentFragment>
  <span
    class="button"
  />
</DocumentFragment>
`;

exports[`Button component should render link using \`href\` 1`] = `
<DocumentFragment>
  <a
    class="button"
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  />
</DocumentFragment>
`;

exports[`Button component should render link using \`to\` 1`] = `
<DocumentFragment>
  <a
    class="button"
    href="#"
  />
</DocumentFragment>
`;
